e2e/native-smoke-test.sh

```
@@ -321,7 +356,224 @@ fork_delete_code=$(curl -sS -o "$fork_delete_body" -w "%{http_code}" -X DELETE \
+
+echo "[12] verify max-file-count enforcement"
+
+FC_DIR="/admin-quota-fc-${TS}"
+drive9_retry fs mkdir "$FC_DIR" >/dev/null
+
+# Set a generous file-count limit so all 5 files are guaranteed to succeed
+set +e
+fc_set_json="$(drive9_ctx admin tenant set-quota \
+  --tenant-id "$TENANT_ID" \
+  --max-file-count 1000 \
+  $ADMIN_FLAGS --json 2>&1)"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)